yarn-project/p2p/src/service/data_store.ts

```diff
@@ -0,0 +1,259 @@
+import type { AztecKVStore, AztecMap } from '@aztec/kv-store';
+
+import { type Batch, type Datastore, Key, type KeyQuery, type Pair, type Query } from 'interface-datastore';
+import type { AwaitIterable } from 'interface-store';
+import filter from 'it-filter';
+import map from 'it-map';
+import sort from 'it-sort';
+import take from 'it-take';
+
+type MemoryItem = {
+  lastAccessedMs: number;
+  data: Uint8Array;
+};
+
+type BatchOp = {
+  type: 'put' | 'del';
+  key: Key;
+  value?: Uint8Array;
+};
+
+export class AztecDatastore implements Datastore {
+  #memoryDatastore: Map<string, MemoryItem>;
```

Comment:
This is implementing an LRU cache over the database. What is this store used for?

One concern I have is that when somebody calls `put`, the data may not be written to disk.